src/NuGet.Jobs.Catalog2Registration/Catalog2RegistrationCommand.cs

```diff
@@ -0,0 +1,127 @@
+            _storageFactory = storageFactory ?? throw new ArgumentNullException(nameof(storageFactory));
+            _handlerFunc = handlerFunc ?? throw new ArgumentNullException(nameof(handlerFunc));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task ExecuteAsync()
+        {
+            await ExecuteAsync(CancellationToken.None);
+        }
+
+        private async Task ExecuteAsync(CancellationToken token)
+        {
+            // Initialize the cursors.
+            ReadCursor backCursor;
+            if (_options.Value.DependencyCursorUrls != null
+                && _options.Value.DependencyCursorUrls.Any())
+            {
+                _logger.LogInformation("Depending on cursors: {DependencyCursorUrls}", _options.Value.DependencyCursorUrls);
+                backCursor = new AggregateCursor(_options
+                    .Value
+                    .DependencyCursorUrls.Select(r => new HttpReadCursor(new Uri(r), _handlerFunc)));
+            }
+            else
+            {
+                _logger.LogInformation("Depending on no cursors, meaning the job will process up to the latest catalog information.");
+                backCursor = MemoryCursor.CreateMax();
+            }
+
+            var frontCursorStorage = _storageFactory.Create();
+            var frontCursorUri = frontCursorStorage.ResolveUri(CursorRelativeUri);
+            var frontCursor = new DurableCursor(frontCursorUri, frontCursorStorage, DateTime.MinValue);
+
+            _logger.LogInformation("Using cursor: {CursurUrl}", frontCursorUri.AbsoluteUri);
+            LogContainerUrl(HiveType.Legacy, c => c.LegacyStorageContainer);
+            LogContainerUrl(HiveType.Gzipped, c => c.GzippedStorageContainer);
+            LogContainerUrl(HiveType.SemVer2, c => c.SemVer2StorageContainer);
+
+            // Optionally create the containers.
+            if (_options.Value.CreateContainers)
+            {
+                await CreateContainerIfExistsAsync(c => c.LegacyStorageContainer);
+                await CreateContainerIfExistsAsync(c => c.GzippedStorageContainer);
+                await CreateContainerIfExistsAsync(c => c.SemVer2StorageContainer);
+            }
+
+            await frontCursor.LoadAsync(token);
+            await backCursor.LoadAsync(token);
+            _logger.LogInformation(
+                "The cursors have been loaded. Front: {FrontCursor}. Back: {BackCursor}.",
+                frontCursor.Value,
+                backCursor.Value);
+
+            // Run the collector.
+            await _collector.RunAsync(
+                frontCursor,
+                backCursor,
+                token);
+        }
+
+        private async Task CreateContainerIfExistsAsync(Func<Catalog2RegistrationConfiguration, string> getContainer)
```

Comment:
This name seems off 😂

```suggestion
        private async Task CreateContainerIfNotExistsAsync(Func<Catalog2RegistrationConfiguration, string> getContainer)
```

Reply:
oops, good catch